pkg/android/get-android-info.go

```diff
@@ -0,0 +1,245 @@
+package android
+
+import (
+	"archive/zip"
+	"bytes"
+	"encoding/xml"
+	"fmt"
+	"github.com/bugsnag/bugsnag-cli/pkg/proto_messages"
+	"google.golang.org/protobuf/proto"
+	"io"
+	"os"
+	"path/filepath"
+)
+
+// Android Manifest reference attriubte IDs
+// https://developer.android.com/reference/android/R.attr#versionCode
+const AndroidVersionCodeId uint32 = 16843291
+
+// https://developer.android.com/reference/android/R.attr#versionName
+const AndroidVersionNameId uint32 = 16843292
+
+type AndroidManifestData struct {
+	XMLName       xml.Name                       `xml:"manifest"`
+	ApplicationId string                         `xml:"package,attr"`
+	VersionCode   string                         `xml:"versionCode,attr"`
+	VersionName   string                         `xml:"versionName,attr"`
+	Application   AndroidManifestApplicationData `xml:"application"`
+}
+
+type AndroidManifestApplicationData struct {
+	XMLName  xml.Name                `xml:"application"`
+	MetaData AndroidManifestMetaData `xml:"meta-data"`
+}
+
+type AndroidManifestMetaData struct {
+	XMLName xml.Name `xml:"meta-data"`
+	Name    []string `xml:"name,attr"`
+	Value   []string `xml:"value,attr"`
+}
+
+func isXMLContent(buffer []byte) bool {
+	xmlHeader := []byte("<manifest xmlns")
+	return bytes.Contains(buffer, xmlHeader)
+}
+
+func isProtobufContent(path string) (bool, error) {
+	// Open the file
+	file, err := os.Open(path)
+	if err != nil {
+		return false, err
+	}
+	defer file.Close()
+
+	// Read the first 8 bytes of the file
+	buffer := make([]byte, 8)
+	_, err = file.Read(buffer)
+	if err != nil {
+		return false, err
+	}
+
+	// Check for the protobuf identifier
+	protobufIdentifier := []byte{0x0A} // Example: Protobuf identifier often starts with 0x0A
+	return bytes.HasPrefix(buffer, protobufIdentifier), nil
+}
+
+// Function to extract the android manifest file from a given aab file
+func GetAndroidManifestFileFromAAB(path string) (string, error) {
+	aabManifestPath := filepath.Join("base", "manifest", "AndroidManifest.xml")
+	outputPath := filepath.Join(path, "..", filepath.Base(aabManifestPath))
+
+	if filepath.Ext(path) == ".aab" {
+		zipData, err := zip.OpenReader(path)
+
+		if err != nil {
+			return "", err
+		}
+
+		defer zipData.Close()
+
+		for _, file := range zipData.File {
+			if file.Name == aabManifestPath {
+				zippedFile, err := file.Open()
+				if err != nil {
+					return "", nil
+				}
+				defer zippedFile.Close()
+
+				destinationFile, err := os.Create(outputPath)
+
+				if err != nil {
+					return "", err
+				}
+
+				defer destinationFile.Close()
+
+				_, err = io.Copy(destinationFile, zippedFile)
+				if err != nil {
+					return "", err
+				}
+
+				return outputPath, nil
+			}
+		}
+	}
+	return "", fmt.Errorf("no aab file")
+}
+
+// GetAndroidXMLData - Pulls information from a human-readable xml file into a struct
+func GetAndroidXMLData(manifestFile string) (*AndroidManifestData, error) {
```

Comment:
Is there any golang naming convention or syntax for marking a function as internal to the package? It would make reading the package a little easier.

Reply:
Usually internal functions are `internalFunction` and public functions are `PublicFunction` or we could move them to an `internal` package so that it becomes `internal.MyFunction`